### DISCUSSION TITLE
Move functions to other *.py files

### DISCUSSION THREAD

**Discussion by GeorgiT-HI** [zauberzeug/nicegui#3890] [Q&A] [answered]
### Question

Hi,
I habe the following app:
```py
from nicegui import ui

def change_icon():
    if my_switch.value == True:
        my_switch.props("icon=play_arrow")
    else:
        my_switch.props("icon=pause")

my_switch = ui.switch("mybutton", on_change=change_icon).props("icon=pause")

ui.run()
```

if I try to move the change_icon function to another file and use

```py
from func import change_icon
```

the 'app' still works but I get an error message: 

> name 'my_switch' is not defined
> Traceback (most recent call last):
>   File "C:\ProgramData\Miniconda3\envs\rfxRoll\Lib\site-packages\nicegui\events.py", line 413, in handle_event
>     result = handler(arguments) if expects_arguments else handler()
>                                                           ^^^^^^^^^
>   File "C:\Users\some.dude\Documents\PythonWebApp\NiceGUI\func.py", line 2, in change_icon
>     if my_switch.value == True:
>        ^^^^^^^^^
> NameError: name 'my_switch' is not defined

I understand why the error is there. But still is there a way to move functions like the one above to other py-files?

Thanks

**Comment by falkoschindler** [comment, marked as answer]
Hi @GeorgiT-HI,

The function shouldn't depend on variables in the global scope. You can avoid it by passing the switch element as an argument:
```py
def change_icon(switch: ui.switch):
    switch.props("icon=play_arrow" if switch.value else "icon=pause")

my_switch = ui.switch("mybutton", on_change=lambda: change_icon(my_switch)).props("icon=pause")
```

Or you define an event handler that accepts `ValueChangeEventArguments`:

```py
def change_icon(e: events.ValueChangeEventArguments):
    e.sender.props("icon=play_arrow" if e.value else "icon=pause")

my_switch = ui.switch("mybutton", on_change=change_icon).props("icon=pause")
```

By the way, you can also use the "checked-icon" and "unchecked-icon" props:
```py
ui.switch("mybutton").props("unchecked-icon=pause checked-icon=play_arrow")
```

**Reply by GeorgiT-HI** [reply]
Thanks, Falko!
